hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -90,6 +137,9 @@ public DatanodeAdminMonitorImpl(
     this.eventQueue = eventQueue;
     this.nodeManager = nodeManager;
     this.replicationManager = replicationManager;
+
+    containerStateByHost = new HashMap<>();
+    pipelinesWaitingToCloseByHost = new HashMap<>();
```

Comment:
Why split the pipelines into a seperate map? It looks like it would be easier overall to have a pipeline count setter on the `ContainerStateInWorkflow` object and just carry the pipeline count around with the containers etc too?

Reply:
Split between replication state and pipelines was for grouping - they are initialized and set in separate parts of the monitor code that resulted in using two separate maps to store the two.  Looking to, as suggested, reuse the `ContainerStateInWorkflow `for the two, perhaps two different setters; one for the replication and the other for pipelines.

Reply:
Combined all metrics collected by host in monitor to `ContainerStateInWorkflow` as suggested.